src/init.cpp

```diff
@@ -1756,6 +1773,8 @@ bool AppInitMain(InitInterfaces& interfaces)
                 // Ensure we are on latest DB version
                 pcustomcsview->SetDbVersion(CCustomCSView::DbVersion);
 
+                EraseLoanAmountsInClosedVaults(*pcustomcsview);
```

Comment:
Instead of startup, what do you think about running it once per block on each block with a close vault tx? More reactive and will keep all nodes in sync.